db/migrate/20250214153835_move_blacklisted_events_to_settings_changes.rb

```diff
@@ -0,0 +1,85 @@
+    "ManageIQ::Providers::Azure::ContainerManager"                   => ["storageAccounts_listKeys_BeginRequest", "storageAccounts_listKeys_EndRequest", "deployments_exportTemplate_BeginRequest", "deployments_exportTemplate_EndRequest"],
+    "ManageIQ::Providers::Amazon::ContainerManager"                  => ["ConfigurationSnapshotDeliveryCompleted", "ConfigurationSnapshotDeliveryStarted", "ConfigurationSnapshotDeliveryFailed"],
+    "ManageIQ::Providers::OracleCloud::CloudManager"                 => ["com.oraclecloud.resourcequeryservice.searchresources", "com.oraclecloud.identitysignon.federatedinteractiveloginattempt", "serviceapi"],
+    "ManageIQ::Providers::Openstack::CloudManager"                   => ["identity.authenticate", "scheduler.run_instance.start", "scheduler.run_instance.scheduled", "scheduler.run_instance.end"],
+    "ManageIQ::Providers::Azure::CloudManager"                       => ["storageAccounts_listKeys_BeginRequest", "storageAccounts_listKeys_EndRequest", "deployments_exportTemplate_BeginRequest", "deployments_exportTemplate_EndRequest"],
+    "ManageIQ::Providers::Amazon::CloudManager"                      => ["ConfigurationSnapshotDeliveryCompleted", "ConfigurationSnapshotDeliveryStarted", "ConfigurationSnapshotDeliveryFailed"],
+    "ManageIQ::Providers::IbmPowerVc::CloudManager"                  => ["identity.authenticate", "scheduler.run_instance.start", "scheduler.run_instance.scheduled", "scheduler.run_instance.end"],
+    "ManageIQ::Providers::IbmCic::CloudManager"                      => ["identity.authenticate", "scheduler.run_instance.start", "scheduler.run_instance.scheduled", "scheduler.run_instance.end"]
+  }.freeze
+
+  # TODO add the rest of the provider settings keys
+  # This is the mapping of BlacklistedEvent#provider_model -> SettingsChange#key
+  PROVIDER_MODEL_TO_SETTINGS_KEY = {
+    "ManageIQ::Providers::Vmware::InfraManager" => "/ems/ems_vmware/blacklisted_event_names"
+  }.freeze
+
+  SETTINGS_KEY_TO_PROVIDER_MODEL = {
+    "/ems/ems_vmware/blacklisted_event_names" => "ManageIQ::Providers::Vmware::InfraManager"
+  }.freeze
+
+  def up
+    say_with_time("Moving BlacklistedEvent records to SettingsChanges") do
+      BlacklistedEvent.in_my_region.where(:system => false, :enabled => true).group_by(&:provider_model).each do |provider_model, blacklisted_events|
+        addition_blacklisted_events = blacklisted_events.pluck(:event_name) - DEFAULT_BLACKLISTED_EVENTS[provider_model]
+        next if addition_blacklisted_events.empty?
+
+        settings_change_key = PROVIDER_MODEL_TO_SETTINGS_KEY[provider_model]
+        next if settings_change_key.nil?
+
+        settings_change = SettingsChange.find_or_initialize_by(:resource_type => "MiqRegion", :resource_id => SettingsChange.my_region_number, :key => settings_change_key)
```

Comment:
The region number doesn't match the MiqRegion `id` column...For example, in region number `0`, the id would probably be `1`, and in region `1` the id would probably be `1000000000001`.

```suggestion
        settings_change = SettingsChange.find_or_initialize_by(:resource_type => "MiqRegion", :resource_id => MiqRegion.my_region.id, :key => settings_change_key)
```

Reply:
You're probably going to have to add an MiqRegion stub for this, annoyingly...you can copy it from another migration.